packages/type-utils/src/predicates.ts

```diff
@@ -8,25 +8,18 @@ const log = debug('typescript-eslint:eslint-plugin:utils:types');
 
 /**
  * Checks if the given type is (or accepts) nullable
- * @param isReceiver true if the type is a receiving type (i.e. the type of a called function's parameter)
  */
-export function isNullableType(
-  type: ts.Type,
-  {
-    isReceiver = false,
-    allowUndefined = true,
-  }: { isReceiver?: boolean; allowUndefined?: boolean } = {},
-): boolean {
+export function isNullableType(type: ts.Type): boolean {
   const flags = getTypeFlags(type);
 
-  if (isReceiver && flags & (ts.TypeFlags.Any | ts.TypeFlags.Unknown)) {
-    return true;
-  }
-
-  if (allowUndefined) {
-    return (flags & (ts.TypeFlags.Null | ts.TypeFlags.Undefined)) !== 0;
-  }
-  return (flags & ts.TypeFlags.Null) !== 0;
+  return (
+    (flags &
+      (ts.TypeFlags.Any |
+        ts.TypeFlags.Unknown |
+        ts.TypeFlags.Null |
+        ts.TypeFlags.Undefined)) !==
+    0
+  );
```

Comment:
[Curious]

what's the difference between this and `isTypeFlagSet`?

```TS
return isTypeFlagSet( 
    ts.TypeFlags.Any |
    ts.TypeFlags.Unknown |
    ts.TypeFlags.Null |
    ts.TypeFlags.Undefined
);
```

Reply:
https://github.com/JoshuaKGoldberg/ts-api-utils/blob/7d515a4a8ff25ec1aeafc3aeeaa874172c69827e/src/flags.ts#L104 -> https://github.com/JoshuaKGoldberg/ts-api-utils/blob/7d515a4a8ff25ec1aeafc3aeeaa874172c69827e/src/flags.ts#L10

Looks like none?

Reply:
I was actually looking here:
https://github.com/typescript-eslint/typescript-eslint/blob/6128a02cb15d500fe22fe265c83e4d7a73ae52c3/packages/type-utils/src/typeFlagUtils.ts#L27

so isn't it more readable? 🤷

Reply:
Yeah, I already did a lot of refactors but didn't go the last mile 🙂